sphinx_gallery/backreferences.py

```diff
@@ -222,7 +238,34 @@ def _make_ref_regex(config=None):
 
 
 def identify_names(script_blocks, ref_regex, global_variables=None, node=""):
-    """Build a codeobj summary by identifying and resolving used names."""
+    """Build a codeobj summary by identifying and resolving used names.
+
+    Parameters
+    ----------
+    script_blocks : list
+        (label, content, line_number)
+        List where each element is a tuple with the label ('text' or 'code'),
+        the corresponding content string of block and the leading line number.
+    ref_regex : str
+        Regex to find references to python objects.
+    example_globals: Optional[Dict[str, Any]]
+        Global variables for examples. Default=None
+    node : ast.Module or str
```

Comment:
When using default 'exec' I think the output is `ast.Module`:
https://greentreesnakes.readthedocs.io/en/latest/tofrom.html#getting-to-and-from-asts